Add support to the multiple file format support feature.

kylasa opened the pull request:
This Pull Request is part of the Distributed Graph Partitioning feature pipeline to support massively large graphs for distributed training. This is to add support to the ParMETIS to read input graph files which are in multiple file format. Here it is assumed that each process is mapped to a input graph file for each of the graph files (nodes, edges, node features and edge features). Each process reads nodes/edges file which is assigned to the current process, using the process-id as suffix string to the appropriate input graph file name. After reading the input graph file, we as

kylasa committed:
Add support for multiple file format input graph files.

This commit is part of the Multiple File format support feature for distributed graph partitioning. This commit enables ParMETIS to read input graph files in multiple file format.

kylasa committed:
Temporary commit - TO BE DELETED.

Only for the purpose of analyzing the code changes for new feature release.

kylasa committed:
Update dglpart2.c

This version is working and tested with mag dataset.

Rhett-Ying commented on programs/dglpart2.c:
```diff
@@ -251,44 +249,79 @@ graph_t *DGLPart_ReadGraph(char *fstem, MPI_Comm comm)
+    char *newstr, *curstr;
 
     chunksize = CHUNKSIZE;
     nchunks = 1+gnedges/chunksize;
```
if `gnedges%chunksize == 0`, does `+1` work as expected? same issue for below `edgecount`.

Rhett-Ying commented on programs/CMakeLists.txt:
```diff
@@ -7,11 +7,12 @@ add_executable(pm_parmetis parmetis.c io.c adaptgraph.c)
 add_executable(pm_pometis pometis.c io.c)
 add_executable(pm_dglpart dglpart.c)
 add_executable(pm_dglpart2 dglpart2.c)
+add_executable(pm_dglpart3 dglpart3.c)
```
where's `dglpart3.c`? the below file should be renamed?

kylasa replied:
dglpart3.c exits in the `programs' folder of the ParMETIS installation.

Rhett-Ying commented on programs/dglpart2.c:
```diff
@@ -55,15 +54,15 @@ int main(int argc, char *argv[])
+  if (argc != 5) {
     if (mype == 0)
-      printf("Usage: %s <fstem> <nparts>\n", argv[0]);
+      printf("Usage: %s <fstem> <nparts> <nodes_file> <edges_file>\n", argv[0]);
```
after reviewing https://github.com/dmlc/dgl/pull/4529, the original `xxx_nodes.txt` and `xxx_edges.txt` have been replaced by `nodes_file` and `edges_files` which includes the all `xxx_nodes/edges.txt` file path. right? If yes, do we need 5 arguments here?

kylasa replied:
No. I think there is some mis-understanding here. I added only two arguments `nfiles` and `efiles` paramters to the existing command. So I guess we only need 4 parameters.

kylasa replied:
About ReadME and example usage, please propose a place where this sample usage command can be placed. But I believe our dgl.doc.ai documentation should have enough documentation about the way we plan to use ParMETIS.

Rhett-Ying commented on programs/dglpart2.c:
```diff
@@ -176,26 +176,24 @@ graph_t *DGLPart_ReadGraph(char *fstem, MPI_Comm comm)
-      printf("ERROR: File '%s' does not exists.\n", filename);
-      ier++;
-    }
+  //sprintf(filename, "%s_nodes_%02d.txt", fstem, mype);
```
I think the content(namely the real path of `xxx_nodes/edges.txt`) in these 2 files should be checked here.

kylasa replied:
Done.

kylasa committed:
Moved Chunked Dataset format support to dglpart3.c.

1. Restored dglpart2.c to its existing code.
2. Moved chunked dataset format support to dglpart3.c

kylasa commented on programs/CMakeLists.txt:
```diff
@@ -7,11 +7,12 @@ add_executable(pm_parmetis parmetis.c io.c adaptgraph.c)
 add_executable(pm_pometis pometis.c io.c)
 add_executable(pm_dglpart dglpart.c)
 add_executable(pm_dglpart2 dglpart2.c)
+add_executable(pm_dglpart3 dglpart3.c)
```
That is the convension which is being followed in this code base. I guess this is something which we cannot change right now.

kylasa committed:
Updated dglpart3.c to address code review comments and also address synthetic graph case.

kylasa committed:
Fixed compilation fix and correctly handling large graphs when reading node weight files from disk

kylasa committed:
Change makefiles to ignore warnings during compilation.

kylasa committed:
Changes made to handle any no. of files when reading edges.
